moon/test_artifacts.py

```diff
@@ -0,0 +1,32 @@
+import unittest
+from unittest.mock import MagicMock
+from pathlib import Path
+import numpy as np
+
+import cv2
+
+from moon.artifacts import ArtifactDetector
+
+curdir = Path(__file__).parent
+
+class ArtifactTest(unittest.TestCase):
+
+    def test_artifacts(self):
+
+        bus = MagicMock()
+        config = {}
+        detector = ArtifactDetector(config, bus)
+        for i in range(len(detector.detectors)):
+            detector.detectors[i]["subsequent_detects_required"] = 0
+
+        with open(str(curdir/'test_data/cube_homebase.jpg'), mode='rb') as img_h:
+            img = img_h.read()
+        self.assertEqual(detector.detect(img, img), [('cubesat', 143, 24, 35, 40, 356), ('homebase', 202, 163, 71, 41, 1594)])
```

Comment:
```
AssertionError: Lists differ: [('cubesat', 143, 24, 35, 40, 355), ('homebase', 202, 164, 70, 40, 1568)] != [('cubesat', 143, 24, 35, 40, 356), ('homebase', 202, 163, 71, 41, 1594)]

First differing element 0:
('cubesat', 143, 24, 35, 40, 355)
('cubesat', 143, 24, 35, 40, 356)

- [('cubesat', 143, 24, 35, 40, 355), ('homebase', 202, 164, 70, 40, 1568)]
?                                 ^                         --------   ^^

+ [('cubesat', 143, 24, 35, 40, 356), ('homebase', 202, 163, 71, 41, 1594)]
?                                 ^                       +++++++ +    ^^
```
i.e. my OpenCV + lib JPEG is slightly different and the results are not identical

Reply:
ok, any suggestion for the right assert that would allow for small differences for these data types?

Reply:
I was going to write some more complicated piece of comparison code, but .. if it works fine on github versions I would go with it (and I may change it in some revised PR with reference to breaking system). I.e. please ignore.

Reply:
As a specific version of the relevant library is required, this is technically a problem with the test platform. May expand to allow was small differences in the future.